internal/manager/config.go

```diff
@@ -101,12 +104,37 @@ type Config struct {
 	// helpful for advanced cases with load-balancers so that the ingress
 	// controller can be gracefully removed/drained from their rotation.
 	TermDelay time.Duration
+
+	flagSet *pflag.FlagSet
 }
 
 // -----------------------------------------------------------------------------
 // Controller Manager - Config - Methods
 // -----------------------------------------------------------------------------
 
+// Validate validates the config. With time this logic may grow to invalidate
+// incorrect configurations.
+func (c *Config) Validate() error {
```

Comment:
Would you mind adding a unit test that would verify the validation rules work as expected? There could be an example valid Config, and test cases could modify its fields to trigger the error.